core/src/main/java/org/springframework/security/authorization/AffirmativeBasedAuthorizationManager.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.springframework.security.access.AccessDecisionVoter;
+import org.springframework.security.access.ConfigAttribute;
+import org.springframework.security.access.SecurityMetadataSource;
+import org.springframework.security.core.Authentication;
+import org.springframework.util.Assert;
+
+/**
+ * Simple concrete implementation of {@link AuthorizationManager} that grants access if
+ * any {@link AccessDecisionVoter} returns an affirmative response.
+ *
+ * @param <T> the type of object that the authorization check is being done one.
+ * @author Evgeniy Cheban
+ */
+public final class AffirmativeBasedAuthorizationManager<T> implements AuthorizationManager<T> {
+
+	private final List<AccessDecisionVoter<?>> decisionVoters;
```

Comment:
I think this implementation requires less changes and we can reuse voters.
I'm not sure how this can be implemented for Method Security, the `AffirmativeBasedAuthorizationManager` is also used in `GlobalMethodSecurityConfiguration`.

Reply:
While I agree that this might be less change, we don't want to introduce a public API that continues to have the same level of complexity as the existing API. The primary goal of this PR is to simplify the authorization API. Generally speaking, we ought to, going forward, avoid using the `ConfigAttribute` approach.

As you mentioned, this is more deeply embedded in method security. I wonder if this class could be package-private to method security configuration, and we can introduce another ticket to continue to rework method security.

Reply:
I forget that `AffirmativeBasedAuthorizationManager` is also used for Web Socket configuration in `AbstractSecurityWebSocketMessageBrokerConfigurer`, maybe for now we can use an adapter for `AccessDecisionManager` and rework Method Security and Web Socket configuration in another ticket?